### PR TITLE
Feature/565

### PR DESCRIPTION
**Pull request by Haytham1337**
Added new functionality to the size page, namely deleting, adding and editing the size:
image
you can delete size:
image
Add and edit size:
image

### CODE REVIEW

File: src/components/forms/size-form/size-form.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        className={commonStyles.sizeTitle}
+      >
+        {config.titles.sizesTitles.sizeAdjustMenu}
+      </Typography>
+      <form className={styles.sizeForm} onSubmit={handleSubmit}>
+        <Grid item xs={12}>
+          <div className={styles.wrapper}>
+            <div className={styles.contentWrapper}>
+              <Paper className={styles.sizeItemAdd}>
+                {sizeList[0].map((item) => (
```
Comment (AndriyLymych) [inline src/components/forms/size-form/size-form.js:94]: use language instead [0]

File: src/components/forms/size-form/size-form.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                >
+                  {Object.values(purposeEnum).map((value) => (
+                    <option key={value} value={value}>
+                      {value}
+                    </option>
+                  ))}
+                </Select>
+              </FormControl>
+              <Paper className={styles.sizeItemAdd}>
+                {sizeList[1].map((item) => (
```
Comment (AndriyLymych) [inline src/components/forms/size-form/size-form.js:146]: same

File: src/components/forms/size-form/size-form.styles.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          paddingRight: 0
+        }
+      }
+    }
+  },
+  sizeItemAdd: {
+    width: '400px',
+    display: 'flex',
+    flexWrap: 'wrap',
+    margin: '20px 0 !important'
```
Comment (AndriyLymych) [inline src/components/forms/size-form/size-form.styles.js:23]: please dont use important!!!!!!

File: src/configs/sizes-enum.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const sizesEnum = {
+  SIZE_L: 'L',
+  SIZE_M: 'M',
+  SIZE_S: 'S'
+};
```
Comment (AndriyLymych) [inline src/configs/sizes-enum.js:5]: add XS, XL, ect....

File: src/pages/sizes/sizes-edit/size-edit.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  const { loading, size } = useSelector(sizeSelector);
+
+  useEffect(() => {
+    dispatch(getSize(id));
+  }, [dispatch, id]);
+
+  if (loading) {
+    return <LoadingBar />;
+  }
+  console.log(size);
```
Comment (AndriyLymych) [inline src/pages/sizes/sizes-edit/size-edit.js:23]: remove console.log

File: src/pages/sizes/sizes.js [hunk 6]
```diff
@@ -30,47 +36,62 @@ const Sizes = () => {
     dispatch(getSizes());
   }, [sizesList]);
 
+  const SizeDeleteHandler = (id) => {
```
Comment (AndriyLymych) [inline src/pages/sizes/sizes.js:39]: srart your naming with small letter

File: src/pages/sizes/sizes.js [hunk 7]
```diff
@@ -30,47 +36,62 @@ const Sizes = () => {
...
     return <LoadingBar />;
   }
-  const sizeItems = map(sizesList, (size) => (
+  const sizeItems = _.map(sizesList, (size) => (
     <TableContainerRow
       showAvatar={false}
       showEdit
       showDelete
-      name='Розмір для сумки'
+      name={size.simpleName[0].value}
```
Comment (AndriyLymych) [inline src/pages/sizes/sizes.js:55]: same

File: src/pages/sizes/sizes.js [hunk 8]
```diff
@@ -30,47 +36,62 @@ const Sizes = () => {
...
       size={size.name}
-      available={size.available ? AVALIABLE_TEXT : UNAVALIABLE_TEXT}
+      available={size.available ? AVAILABLE_TEXT : UNAVAILABLE_TEXT}
       id={size._id}
       key={size._id}
+      deleteHandler={() => {
+        SizeDeleteHandler(size._id);
+      }}
+      editHandler={() => {
+        dispatch(push(`/sizes/${size._id}`));
```
Comment (AndriyLymych) [inline src/pages/sizes/sizes.js:64]: use rout from config

File: src/redux/sizes/sizes.operations.js [hunk 9]
```diff
@@ -20,3 +29,135 @@ export const getAllSizes = async () => {
...
+      }
+    `,
+    fetchPolicy: 'no-cache'
+  });
+
+  return result.data.getSizeById;
+};
+
+export const addSize = async (size) => {
+  const token = getFromLocalStorage(config.tokenName);
```
Comment (AndriyLymych) [inline src/redux/sizes/sizes.operations.js:66]: push enter

File: src/redux/sizes/sizes.operations.js [hunk 10]
```diff
@@ -20,3 +29,135 @@ export const getAllSizes = async () => {
...
+      `${result.data.addSize.statusCode} ${[result.data.addSize.message]}`
+    );
+  }
+
+  return result.data.addSize;
+};
+
+export const updateSize = async (id, size) => {
+  const token = getFromLocalStorage(config.tokenName);
+  const result = await client.mutate({
```
Comment (AndriyLymych) [inline src/redux/sizes/sizes.operations.js:101]: same

File: src/components/forms/size-form/size-form.js (hunk 1 above)
Comment (Haytham1337) [reply src/components/forms/size-form/size-form.js]: solved

File: src/components/forms/size-form/size-form.js (hunk 2 above)
Comment (Haytham1337) [reply src/components/forms/size-form/size-form.js]: solved

File: src/components/forms/size-form/size-form.styles.js (hunk 3 above)
Comment (Haytham1337) [reply src/components/forms/size-form/size-form.styles.js]: removed

File: src/configs/sizes-enum.js (hunk 4 above)
Comment (Haytham1337) [reply src/configs/sizes-enum.js:8]: add more values

File: src/pages/sizes/sizes-edit/size-edit.js (hunk 5 above)
Comment (Haytham1337) [reply src/pages/sizes/sizes-edit/size-edit.js]: removed consle.log

File: src/pages/sizes/sizes.js (hunk 6 above)
Comment (Haytham1337) [reply src/pages/sizes/sizes.js]: rename

File: src/pages/sizes/sizes.js (hunk 7 above)
Comment (Haytham1337) [reply src/pages/sizes/sizes.js:55]: solved

File: src/pages/sizes/sizes.js (hunk 8 above)
Comment (Haytham1337) [reply src/pages/sizes/sizes.js]: done

File: src/redux/sizes/sizes.operations.js (hunk 9 above)
Comment (Haytham1337) [reply src/redux/sizes/sizes.operations.js:61]: done

File: src/redux/sizes/sizes.operations.js (hunk 10 above)
Comment (Haytham1337) [reply src/redux/sizes/sizes.operations.js:96]: done

**Review by AndriyLymych — APPROVED** [review]

**Review by jul4ajka — APPROVED** [review]

**Review by Kiyooooko — APPROVED** [review]

**Pull request merged**